daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala

```diff
@@ -441,11 +434,8 @@ object ScenarioLedger {
                       val mbNewCache2 = nc.key match {
                         case None => Right(newCache1)
                         case Some(keyWithMaintainers) =>
-                          val gk = GlobalKey(
-                            nc.coinst.template,
-                            // FIXME: we probably should never crash here !
-                            assertNoContractId(keyWithMaintainers.key),
-                          )
+                          // FIXME: we probably should never crash here !
+                          val gk = GlobalKey.assertBuild(nc.coinst.template, keyWithMaintainers.key)
```

Comment:
How about turning these `FIXMEs` into failfast checks?

Reply:
@fabiotudone-da what exactly are you suggesting? A crash sounds like it fails fast :slightly_smiling_face: 

Either way, this hasn’t changed in this PR so I wouldn’t block it on this.

Reply:
Absolutely, so I propose to either remove the FIXMEs altogether, if the assert message (+stack) is dev friendly enough, or catching and re-throwing with a more specific/contextualized error.